framework/src/meshgenerators/BreakMeshByBlockGenerator.C

```diff
@@ -48,21 +52,44 @@ BreakMeshByBlockGenerator::validParams()
 BreakMeshByBlockGenerator::BreakMeshByBlockGenerator(const InputParameters & parameters)
   : BreakMeshByBlockGeneratorBase(parameters),
     _input(getMesh("input")),
-    _block(parameters.isParamSetByUser("block") ? getParam<std::vector<SubdomainID>>("block")
-                                                : std::vector<SubdomainID>()),
-    _block_set(_block.begin(), _block.end()),
-    _block_restricted(parameters.isParamSetByUser("block")),
+    _block_pairs_restricted(parameters.isParamSetByUser("block_pairs")),
+    _surrounding_blocks_restricted(parameters.isParamSetByUser("surrounding_blocks")),
     _add_transition_interface(getParam<bool>("add_transition_interface")),
     _split_transition_interface(getParam<bool>("split_transition_interface")),
-    _interface_transition_name(getParam<BoundaryName>("interface_transition_name"))
+    _interface_transition_name(getParam<BoundaryName>("interface_transition_name")),
+    _add_interface_on_two_sides(getParam<bool>("add_interface_on_two_sides"))
 {
+  if (_block_pairs_restricted && _surrounding_blocks_restricted)
+    paramError("'surrounding_blocks' and 'block_pairs' can not be used at the same time.");
+
+  if (_block_pairs_restricted)
+  {
+    for (auto block_pair : getParam<std::vector<std::vector<SubdomainID>>>("block_pairs"))
```

Comment:
We don't have a `std::vector<std::pair<>>` parameter support? Well, then this will have to do.

Reply:
I guess we don't have the support yet. :-)